src/CashuWallet.ts

```diff
@@ -78,16 +78,14 @@ class CashuWallet {
     }
 
     async receive(encodedToken: string): Promise<Array<Proof>> {
-        const { proofs, mints } = getDecodedProofs(encodedToken)
-        const amount = proofs.reduce((total, curr) => {
-            return total + curr.amount
-        }, 0)
+        const { proofs } = getDecodedProofs(encodedToken)
+        const amount = proofs.reduce((total, curr) => total + curr.amount, 0)
         const { payload, amount1BlindedMessages, amount2BlindedMessages } = await this.createSplitPayload(0, amount, proofs)
         const { fst, snd } = await this.mint.split(payload)
+        if(!fst || !snd) { return [] }
```

Comment:
not sure about this one. maybe better to split this up into two different statements, and only set the given proof array to empty if needed and return at the end.  thoughts?